### PR TITLE
feat: typescript migration of alert and icon

### PR DESCRIPTION
**Pull request by DorianMaliszewski**
## Type

- Migration

#### What is expected?

Migration of Alert and Icon

#### The following changes where made:

1. start from my first pr

2. Migrate Alert

3. Migrate Icon

### CODE REVIEW

File: src/types.ts [hunk 1]
```diff
@@ -0,0 +1,13 @@
+export type XStyledProps = {
+  as?: string | unknown
```
Comment (matthprost) [inline src/types.ts:2]: Why unknown ? Shouldn't it be string only ?

**Review by matthprost — CHANGES_REQUESTED** [review]

File: src/components/ActionBar/index.tsx [hunk 2]
```diff
@@ -26,18 +26,18 @@ const StyledBox = styled(Box)<{ rank: number }>`
 `
 
 type ActionBarProps = {
-  children: React.ReactNode
+  children: ReactNode
```
Comment (adriengibrat) [inline src/components/ActionBar/index.tsx:29]:
```suggestion
```
`FunctionComponent` (or it's alias `FC`) implies `children`: no need to specify it in Props type definition

File: src/components/Avatar/index.tsx [hunk 3]
```diff
@@ -1,11 +1,12 @@
 import styled from '@emotion/styled'
 import PropTypes from 'prop-types'
```
Comment (adriengibrat) [inline src/components/Avatar/index.tsx:2]: migrate to Ts & keep proptypes ?

Maybe https://openbase.com/js/babel-plugin-typescript-to-proptypes could help DRY ;)

File: src/components/ActionBar/index.tsx (hunk 2 above)
Comment (chambo-e) [reply src/components/ActionBar/index.tsx:29]: There is a bug in react types which currently force us to specify children in props

I submitted a PR to correct this https://github.com/DefinitelyTyped/DefinitelyTyped/pull/54955 :)

File: src/components/Avatar/index.tsx (hunk 3 above)
Comment (chambo-e) [reply src/components/Avatar/index.tsx:2]: You're right, we've decided to keep them both while migrating but an automated way to convert types into proptypes would be of great value when completed :)

File: src/components/ActionBar/index.tsx (hunk 2 above)
Comment (chambo-e) [reply src/components/ActionBar/index.tsx:29]: I'm not gonna have it ready anytime soon, I think we'll have to move forward without it for now 😢

**Review by matthprost — APPROVED** [review]

**Review by philibea — APPROVED** [review]

**Pull request merged**